### ISSUE TITLE
pdf embedded

### ISSUE THREAD

**Issue by vivaleon2** [wkhtmltopdf/wkhtmltopdf#3345]
Please, I need solution my issue:
My goal is generate a pdf with wkhtmltopdf from an html with others pdf files embebed. I've tried with all my knowns ways to include a pdf file in html, but when I run wkhtmltopdf always blank page is generated. Instead If I use jpg files, WK work well. But I need same with pdf files, and can't convert to image previously.

This sample code shows 3 ways to embed a pdf into html:

`<body>`
`<iframe src="test.pdf"></iframe>`
`<object data="test.PDF#page=1" type="application/pdf" width="100%" height="100%"></object>`
`<embed src="test.pdf#page=1" type="application/pdf" width="100%" height="300%">`
`</body>`

My simple query : wkhtmltopdf test.HTML result.PDF
on Windows 10 and Windows Server 2012, version 0.12.4.0, and all files and executions inside bin folder.

How must I include the "local" pdf file in my html and wkhtmltopdf undesrtand well?.
Thanks in advance

**Comment by willemmulder** [comment]
If this is an issue with wkhtmltopdf (i.e. it should work, but it doesn't), then provide a bit more background on why you think it should work.

If this is an open question on how to embed pdf files in HTML files, then ask on e.g. https://stackoverflow.com or the mailing group.

**Comment by PhilterPaper** [comment]
A _browser_ engine, such as WebKit, does not natively render PDFs. It uses some sort of plug-in, such as Adobe Reader. Does anyone know if WebKit (used by wkHTMLtoPDF) is able to load a PDF plug-in? 

What is your ultimate objective? Do you have several PDFs that you want to combine into one? It might be easier in that case to use a PDF-merge tool (there are many). If you're looking to add some content (text, images, etc.) along with the PDFs, that would be more difficult. Perhaps you could convert the HTML to PDF(s) and then merge those file(s) with the PDFs?

**Comment by willemmulder** [comment]
If you really want to render PDFs in the browser yourself, there is PDF.js if you want to stretch it: https://mozilla.github.io/pdf.js/

It's a PDF rendering engine from Mozilla for use in the browser. It's the default PDF renderer for Firefox. I'm not sure if wkhtmltopdf supports it fully, but you can always try it out :-)

**Comment by vivaleon2** [comment]
Thanks your interest. 
I have html files with pdf embedded inside. And I need generate a pdf file from this html. If I embed images, wkhtmltopdf works fine. But when I embed pdf files, the execution of wkhtmltopdf results in white pages. Without errors in return wk command.
I know how embed pdf into html, in attached sample, there are three ways. And any works fine with wkhtmltopdf. 
[pdfembedsample.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/797380/pdfembedsample.zip)

Is it an Issue of wkhtmltopdf? 
Thanks in advance
